core/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -307,10 +307,16 @@ public static GetResult fromXContentEmbedded(XContentParser parser) throws IOExc
                         fields.put(getField.getName(), getField);
                     }
                 } else {
-                    throwUnknownField(currentFieldName, parser.getTokenLocation());
+                    parser.skipChildren(); // skip potential inner objects for forward compatibility
                 }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                parser.skipChildren(); // skip potential inner arrays for forward compatibility
             }
         }
+
+        if (found == null) {
```

Comment:
For the Update API, a 404 is returned with an exception in the response body. The high level rest client doesn't ignore 404s but it's okay because the response body will always be an exception.

The Get API is a bit different, when the document exists it returns a 404 but with a response body that is not an exception. So the client has to try to parse that doc, fails, and then parses it as an exception and got the right result.